dbms/src/Storages/DeltaMerge/File/DMFileWriter.h

```diff
@@ -70,39 +69,31 @@ class DMFileWriter
                                  ? std::unique_ptr<WriteBuffer>(new CompressedWriteBuffer<false>(*plain_file, compression_settings))
                                  : std::unique_ptr<WriteBuffer>(new CompressedWriteBuffer<true>(*plain_file, compression_settings)))
             , minmaxes(do_index ? std::make_shared<MinMaxIndex>(*type) : nullptr)
-            , mark_file(WriteBufferByFileProviderBuilder(
-                            dmfile->configuration.has_value(),
-                            file_provider,
-                            dmfile->colMarkPath(file_base_name),
-                            dmfile->encryptionMarkPath(file_base_name),
-                            false,
-                            write_limiter_)
-                            .with_checksum_algorithm(detail::getAlgorithmOrNone(*dmfile))
-                            .with_checksum_frame_size(detail::getFrameSizeOrDefault(*dmfile))
-                            .build())
+            , marks(std::make_shared<MarksInCompressedFile>())
```

Comment:
Only initialize `marks` when `dmfile->useMetaV2()` is true.

Reply:
done